JS/Lesson26/savas/script.js

```diff
@@ -6,9 +6,110 @@ p (.post-body)
 */
 
 const URL = 'https://jsonplaceholder.typicode.com/posts';
+
+document.addEventListener("DOMContentLoaded", function() {
+    const getPostsButton = document.getElementById("get-posts");
+    if (getPostsButton) {
+      getPostsButton.addEventListener("click", fetchPosts);
+    }
+  
+    const createPostButton = document.getElementById("create-a-post");
+    if (createPostButton) {
+      createPostButton.addEventListener("click", function() {
+        window.location.href = "create-a-post.html";
+      });
+    }
+  
+    const clearPostsButton = document.getElementById("clear-posts");
+    if (clearPostsButton) {
+      clearPostsButton.addEventListener("click", function() {
+        const postsContainer = document.getElementById("posts-container");
+        if (postsContainer) {
+          postsContainer.innerHTML = "";
+        }
+      });
+    }
+  
+    const submitButton = document.getElementById("submit");
+    if (submitButton) {
+      submitButton.addEventListener("click", createANewPost);
+    }
+  });
+
+//-----------------FUNCTION FOR GET AND EDIT POSTS-----------------
+
 function fetchPosts() {
+    fetch(URL)
+    .then(response => response.json())
+    .then(json => {
+        let posts = "";
+
+        json.forEach(element => {
+            posts += `<div class="post">
+                <h2 class="post-title"><span>Post ${element.id}</span> - ${element.title}</h2>
+                <p class="post-body">${element.body}</p>
+                <button class="button--edit" data-id="${element.id}">Edit Post</button>
+                <button class="button--danger" data-id="${element.id}">Delete Post</button>
+            </div>`;
+        });
+
+        document.getElementById("posts-container").innerHTML = posts;
```

Comment:
innerHTML is an "unsafe" method. Try to create elements one by one and append them to each other instead. This is a longer approach but it is a safer choice.
E.g.
const heading = document.createElement('h2');
...

Actually,  `<span>Post ${element.id}</span> - ${element.title}` this is the only code that I'd use innerHTML for:
heading.innerHTML = `<span>Post ${element.id}</span> - ${element.title}`;

For other things I'd set attributes and innerText one by one.

Reply:
Anna, I don't understand why you wrote this comment. For the id, title and body of the elements above? (line between 33 and 38)

Reply:
This comment is about this code:
```
<div class="post">
                <h2 class="post-title"><span>Post ${element.id}</span> - ${element.title}</h2>
                <p class="post-body">${element.body}</p>
                <button class="button--edit" data-id="${element.id}">Edit Post</button>
                <button class="button--danger" data-id="${element.id}">Delete Post</button>
            </div>
```

Instead of adding these elements as innerHTML, it's better to create them as I mentioned above